### PR TITLE
CY-1872 Add node_id for Rabbit and Postgres nodes in the config

### PR DESCRIPTION
**Pull request by inbala**
* Change the format of db nodes in the config.yaml, it is a dict
  instead of a list and it includes the hostname, ip and node_id of a db node.

* Change the format of rabbit nodes in the config.yaml, add node_id and move
  the networks under 'networks' key.

* Insert the db nodes info to the DB.

* Set the hostname of the db and rabbit nodes in all-in-one manager to be the
  hostname of the manager.

* Take the manager node_id from the config of the reporter.

* Set the node_id of the db and rabbit nodes in all-in-one manager to be the
  node_id of the manager.

* If it is not all-in-one and there is no node_id configured in the rabbit nodes
  section, we assume it is an external rabbit (user-provided).

### CODE REVIEW

File: .circleci/cluster/manager1_config.yaml [hunk 1]
```diff
@@ -10,7 +10,9 @@ rabbitmq:
   ca_path: /etc/cloudify/ca.pem
   cluster_members:
     rabbit1:
-      default: QUEUE_IP
+      node_id: 'rabbit1_node_id'
```
Comment (geokala) [inline .circleci/cluster/manager1_config.yaml]: Is there a reason not to have the node id as the key, rather than having a separate node_id key:value pair under the dict??

File: .circleci/cluster/manager1_config.yaml [hunk 2]
```diff
@@ -19,9 +21,15 @@ postgresql_server:
   postgres_password: 'somesecretvalue123!'
   cluster:
     nodes:
-      - DB1_IP
-      - DB2_IP
-      - DB3_IP
+      db_1:
+        ip: DB1_IP
+        node_id: 'db1_node_id'
```
Comment (geokala) [inline .circleci/cluster/manager1_config.yaml]: Same question as for rabbit.

File: cfy_manager/components/rabbitmq/rabbitmq.py [hunk 3]
```diff
@@ -409,7 +409,9 @@ def _generate_rabbitmq_certs(self):
             # anything consuming this value will get the path to the cert
             # that will allow them to trust the broker.
             config[RABBITMQ]['ca_path'] = config[RABBITMQ]['cert_path']
-        if len(config[RABBITMQ]['cluster_members']) > 1:
+
+        cluster_members = config[RABBITMQ]['cluster_members']
```
Comment (geokala) [inline cfy_manager/components/rabbitmq/rabbitmq.py]: What's the purpose of this change?

File: cfy_manager/components/restservice/config/haproxy.cfg [hunk 4]
```diff
@@ -18,7 +18,7 @@ listen postgres
     option httpchk
     http-check expect status 200
     default-server inter 3s fall 3 rise 2 on-marked-down shutdown-sessions
-{%- for node in postgresql_server.cluster.nodes %}
-    server postgresql_{{ node }}_5432 {{ node }}:5432 maxconn 100 check check-ssl port 8008 ca-file /etc/haproxy/ca.crt
+{%- for name, node in postgresql_server.cluster.nodes.iteritems() %}
```
Comment (geokala) [inline cfy_manager/components/restservice/config/haproxy.cfg]: If we're discarding the name it'd be better to do:
for node in postgresql_server.cluster.nodes.values()

File: cfy_manager/components/restservice/db.py [hunk 5]
```diff
@@ -35,17 +36,19 @@
 from ..service_components import DATABASE_SERVICE
 from ..service_names import (
     MANAGER,
-    POSTGRESQL_CLIENT,
     RABBITMQ,
     RESTSERVICE,
+    POSTGRESQL_CLIENT,
```
Comment (geokala) [inline cfy_manager/components/restservice/db.py]: Why not add POSTGRESQL_SERVER in alphabetical order the way POSTGRESQL_CLIENT was before?

File: config.yaml [hunk 6]
```diff
@@ -92,16 +92,20 @@ rabbitmq:
...
   # The 'default' network IP must be set for each member.
-  # If installing a single external rabbit (not intended to be part of a cluster),
-  # or an all-in-one manager this section can be left blank.
+  # If installing an all-in-one manager this section can be left blank.
   # Example:
   # cluster_members:
   #   <hostname of rabbit node>:
-  #     default: <ip of rabbit node> (not needed if node name is resolvable via DNS)
-  #     <other network name>: <ip for this node on 'other network'>
+  #     node_id: <Cloudify's auto-generated id of the rabbit node> (only needed on
```
Comment (geokala) [inline config.yaml]: If it's auto-generated and not user controlled, why is it in the config?

File: config.yaml [hunk 7]
```diff
@@ -234,12 +238,19 @@ postgresql_server:
...
-    # The nodes must be provided as IPs, e.g.
+    # The nodes must be provided with IPs, e.g.
     # nodes:
-    #   - 192.0.2.1
-    #   - 192.0.2.2
-    #   - 192.0.2.3
-    nodes: []
+    #   <hostname of postgres node>:
+    #     ip: <ip of postgres node>
+    #     node_id: <Cloudify's auto-generated id of the postgres node> (only needed on
```
Comment (geokala) [inline config.yaml]: Same question as for rabbit node_id

File: .circleci/cluster/manager1_config.yaml (hunk 1 above)
Comment (inbala) [reply .circleci/cluster/manager1_config.yaml]: Yes, I want the hostname as the key.

File: cfy_manager/components/rabbitmq/rabbitmq.py (hunk 3 above)
Comment (inbala) [reply cfy_manager/components/rabbitmq/rabbitmq.py]: I changed it :)
I think it was because line 422 was too long, but now it is fine.

File: .circleci/cluster/manager1_config.yaml (hunk 2 above)
Comment (inbala) [reply .circleci/cluster/manager1_config.yaml]: I want to have also the hostname, I'll use the hostname in the response of the cluster status.

File: config.yaml (hunk 6 above)
Comment (inbala) [reply config.yaml]: Good question, I'm not sure how to phrase it right.
The node_id is auto-generated when installing the status reporter on the rabbit/postgres node and the user has to take it and add it here as he adds the IP and hostname.

File: cfy_manager/components/postgresql_server/postgresql_server.py [hunk 8]
```diff
@@ -741,10 +744,10 @@ def _create_patroni_config(self, patroni_config_path):
                 'password': pgsrv['cluster']['etcd']['patroni_password']
             },
         }
-        for node in pgsrv['cluster']['nodes']:
+        for node in pgsrv['cluster']['nodes'].values():
```
Comment (tyacbovi) [inline cfy_manager/components/postgresql_server/postgresql_server.py]: maybe pgsrv -> db_server/postgres_server

Comment (inbala) [reply cfy_manager/components/postgresql_server/postgresql_server.py]: I agree it is not a good name, I assume it was chosen so lines like `676` and more will be one liner.

Comment (inbala) [reply cfy_manager/components/postgresql_server/postgresql_server.py]: I agree it is not a good name, I assume it was chosen so lines like 676 and more will be one liner.
